Add the ability to use game specific settings

pivotiiii opened the pull request:
This basically applies the --cfgpath="path" argument on game launch if the feature is enabled, allowing the usage of different settings for individual games.

pivotiiii committed:
Add the ability to use game specific settings.

By selecting your configs folder you can use and tweaks settings on a per game basis. For this to work, a folder with the same name as the rom has to exist in the "Documents/PCSX2/configs" folder. 

Also fixed a typo in GameIndex.txt

pivotiiii committed:
Update README.md

pivotiiii committed:
Clarified usage.

pivotiiii committed:
Update README.md

NoShotz commented:
Found an issue, if the game name in GOG has a character that can't be used in a folder name such as a : it won't work. It's also adding quotes where there shouldn't be, https://i.imgur.com/JmmOn0i.png

There shouldn't be quotes around --cfgpath=E:/Games/Emulators/PCSX2/configs/Sly 2: Band of Thieves

It should be --cfgpath="E:/Games/Emulators/PCSX2/configs/Sly 2 Band of Thieves" as :'s can't be used in folder names.

pivotiiii committed:
special characters/ no new folders

The config folders where searched for by the gog game name, which can contain special characters that are not allowed in folder names. This commit fixes that by using the rom name as the folder name, since they are bound by the same restrictions.
I also fixed the plugin accidentally creating a new folder for every game that did not have one already. Only games the user actually wants to use individual settings with should use one.

pivotiiii commented:
Thanks for the heads up, I fixed the special characters by using the rom name and not the game name, which I should have done in the first place. The quotes are around the whole argument and not just the path, because PCSX2 was acting weird with escaped quotes in the path. It still works this way.

NoShotz commented:
It's working great now.
